roles/confluent.kafka_connect/library/confluent_connectors.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Confluent Inc
+
+ANSIBLE_METADATA = {
+    'metadata_version': '0.9',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: confluent_connectors
+
+short_description: This module allows setting up Confluent connectors from Ansible.
+
+version_added: "2.4"
+
+description:
+    - "This module allows setting up Confluent connectors from Ansible. It registers the new ones,
+    updates the existing ones and removes the deleted ones."
+
+options:
+    connect_url:
+        description:
+            - URL of the Connect REST server to use to add/edit connectors
+        required: true
+    active_connectors:
+        description:
+            - Dict of active connectors (each connector object must have a 'name' and a 'config' field)
+        required: true
+
+author:
+    - Laurent Domenech-Cabaud (@ldom)
+'''
+
+EXAMPLES = '''
+- connect_url: {{kafka_connect_http_protocol}}://0.0.0.0:{{kafka_connect_rest_port}}/connectors
+  active_connectors: [{"name": "test-6-sink", "config": { .../... }},{"name": "test-5-sink", "config": { .../... }}]
+'''
+
+RETURN = '''
+message:
+    description: The output message that the module generates
+    type: str
+    returned: always
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+
+
+def get_current_connectors(connect_url):
+    try:
+        res = open_url(connect_url)
+        return json.loads(res.read())
+    except urllib_error.HTTPError as e:
+        if e.code != 404:
+            raise
+        return []
+
+def remove_connector(connect_url, name):
+    url = "{}/{}".format(connect_url, name)
+    r = open_url(method='DELETE', url=url)
+    return r.getcode() == 200
+
+def install_new_connector(connect_url, name, config):
+    data = json.dumps({'name': name, 'config': config})
+    headers = {'Content-Type': 'application/json'}
+    r = open_url(method='POST', url=connect_url, data=data, headers=headers)
+    return r.getcode() in (200, 201, 409)
+
+def update_existing_connector(connect_url, name, config):
+    url = "{}/{}/config".format(connect_url, name)
+    restart_url = "{}/{}/restart".format(connect_url, name)
+
+    res = open_url(url)
+    current_config = json.loads(res.read())
```

Comment:
🤔 Surprised `open_url` doesn't have a `.json()` method like `requests` does

Reply:
Yeah, we have to use the classes provided by Ansible because they've added a layer of compatibility so that you can run the scripts on python 2 and 3. Their classes are a little bit limited...